File: 71.java
import java.util.Stack;

class Solution {
    public String simplifyPath(String path) {
        Stack<String> stack = new Stack<>();
        for (String i : path.split("/")) {
            if (i.isEmpty() || ".".equals(i))
                continue;
            if ("..".equals(i)) {
                if (!stack.isEmpty())
                    stack.pop();
            } else
                stack.push(i);
        }
        return "/" + String.join("/", stack);
    }
}

File: Python/71.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        for step in path.split("/"):
            match step:
                case "" | ".":
                    continue
                case "..":
                    if stack:
                        stack.pop()
                case _:
                    stack.append(step)

        return "/" + "/".join(stack)
